Drop-In/src/main/java/com/braintreepayments/api/DropInActivity.java

```diff
@@ -44,10 +45,29 @@
 
     static final String EXTRA_PAYMENT_METHOD_NONCES = "com.braintreepayments.api.EXTRA_PAYMENT_METHOD_NONCES";
 
-    private DropInViewModel dropInViewModel;
+    @VisibleForTesting
+    DropInViewModel dropInViewModel;
     ActionBar actionBar;
     private FragmentContainerView fragmentContainerView;
 
+    @Override
+    protected void onResume() {
+        super.onResume();
+
+        getDropInClient().deliverBrowserSwitchResult(this, new DropInResultCallback() {
+            @Override
+            public void onResult(@Nullable DropInResult dropInResult, @Nullable Exception error) {
+                if (dropInResult != null) {
+                    finishWithDropInResult(dropInResult);
+                } else if (error instanceof UserCanceledException) {
+                    dropInViewModel.setIsLoading(false);
```

Comment:
Handling cancellation by setting `isLoading` to false in `dropInViewModel`. `CardDetailsFragment` is resumed on 3DS cancellation and observes the change in `isLoading` to set the animated button view to clickable again.

Reply:
I feel like our loading logic is going to evolve over time since there are a few more structural PRs we can do to clean up some of the event code. I wonder if it's something we can tackle in another PR since it's technically polish.
🇵🇱

Reply:
This change was based on the assumption that the flow should work the same way as the current version, so we need to reset the loading status of the submit button when a user cancels.  I think that change makes sense in this PR if we are going with the previous version flow. If we want to dismiss Drop in all together on 3DS cancel, I will take out the loading logic.

Reply:
Yeah we might be able to tackle this in another PR, but I'm starting to feel like there will be more than one loader as we get closer to GA. I think we're overloading the `DropInViewModel#isLoading` property a lot; it isn't clear which loader that value binds to at the moment.

Reply:
I agree with Steven that it is getting a little confusing.

> CardDetailsFragment is resumed on 3DS cancellation and observes the change in isLoading to set the animated button view to clickable again.

This isn't immediately obvious, without digging into CardDetailsFragment code. We could have an additional property (instead of re-using `isLoading`) with a more distinct purpose/name solely for signaling the submit button?